### PR TITLE
7.c :reverse of a number with overflow check added

### PR DESCRIPTION
**Pull request by SuranjanDaw**

### CODE REVIEW

File: leetcode/src/7.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+int reverse(int x){
```
Comment (danghai) [inline leetcode/src/7.c]: So many checking here. I have a good solution for it:
```
int reverse(int x){
    long res = 0;
    while (x != 0)
    {
        int tail = x % 10;
        res = res * 10 + tail;
        /* Check overflow */
        if (res > INT_MAX || res < INT_MIN)
            return 0;
        x = x / 10;
    }
    return (int)res;
}
```

File: leetcode/src/9.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
+
```
Comment (danghai) [inline leetcode/src/9.c]: a better solution:
```
int reverse(int x){
    long res = 0;
    while (x != 0)
    {
        int tail = x % 10;
        res = res * 10 + tail;
        /* Check overflow */
        if (res > INT_MAX || res < INT_MIN)
            return 0;
        x = x / 10;
    }
    return (int)res;
}

bool isPalindrome(int x) {
    if (x < 0)
        return 0;
    else
        return x == reverse(x);
}
```

File: leetcode/src/7.c (hunk 1 above)
Comment (SuranjanDaw) [reply leetcode/src/7.c]: Here you are using long int. What if the system we are using, supports up to 32-bit? What if the OS does not support long int?

Comment (danghai) [reply leetcode/src/7.c]: The question is not limit only for 32-bit.

Comment (SuranjanDaw) [reply leetcode/src/7.c]: Yeah if we don't put that restriction then it will work. So do you want me to change and re-push?

Comment (danghai) [reply leetcode/src/7.c]: You can put the comment on your way, it works only for 32-bit system. And then adding my solution below. Thank you

Comment (SuranjanDaw) [reply leetcode/src/7.c]: Committed the changes. Thanks

**Pull request merged**